### PR TITLE
[1LP][RFR] reworking requests

### PR DESCRIPTION
**Pull request by lkhomenk**
__rework__ for requests.
Sometimes we just need to find Request in the table - without waiting it to finish etc

### CODE REVIEW

File: cfme/services/requests.py [hunk 1]
```diff
@@ -217,28 +194,40 @@ def debug_requests():
...
     """
     navigate_to(Request, 'All')
     for page in paginator.pages():
-        try:
-            # found the row!
-            row, = fields.request_list.find_rows_by_cells(cells)
-            return True
-        except ValueError:
+        # We check only for the SplitTable. Can't think of better detection.
+        if version.current_version() < "5.5.0.8"\
```
Comment (psav) [inline cfme/services/requests.py]: **REQUIRED:** As we are reworking this, let's remove the 5.5 logic as 5.5 is no longer supported

Comment (lkhomenk) [inline cfme/services/requests.py]: removed

File: cfme/tests/services/test_provision_stack.py [hunk 2]
```diff
@@ -200,7 +200,6 @@ def test_remove_template_provisioning(provider, provisioning, catalog, catalog_i
     template.delete()
     row_description = 'Provisioning Service [{}] from [{}]'.format(item_name, item_name)
     cells = {'Description': row_description}
-    wait_for(lambda: requests.find_request(cells), num_sec=500, delay=20)
```
Comment (lkhomenk) [inline cfme/tests/services/test_provision_stack.py]: added if_exists for such cases. but this particular doesn't need it at all - wait_for_request does all the same. so saving time a bit

File: cfme/services/requests.py [hunk 3]
```diff
@@ -217,28 +194,38 @@ def debug_requests():
...
             continue
+        elif len(results) > 1:
+            raise RequestException(
+                'Multiple requests with matching content found - be more specific!'
+            )
+        else:
+            # found the row!
+            row = results[0]
+            logger.debug(' Request Message: %s', row.last_message.text)
+            break
```
Comment (mshriver) [inline cfme/services/requests.py]: **Required**
Return here instead of breaking

File: cfme/services/requests.py [hunk 4]
```diff
@@ -250,19 +237,12 @@ def go_to_request(cells, partial_check=False):
...
-            # row not on this page, assume it has yet to appear
-            # it might be nice to add an option to fail at this point
-            continue
-    else:
-        # Request not found at all, can't continue
+    try:
+        row = find_request(cells, partial_check)
+        sel.click(row)
+        return True
+    except Exception as e:
```
Comment (mshriver) [inline cfme/services/requests.py]: **Required**
Its not clear which of these calls may result in an exception that gets put in the log and ignored.

If the row isn't found, find_request returns false, which means its the sel.click throwing an exception when really the failure was in find_request.

I don't think all exceptions should be hidden here, and the point of failure should be more clear.

Consider something structured more like:
```python
row = find_request(cells, partial_check)
if row:
    sel.click(row)
    return True
else:
    return False
```

**Comment by lkhomenk** [comment]
widgetastic and navmazing will come apart from this PR

**Comment by mshriver** [comment]
PRT results have failures but so far look unrelated to your changes, I'll merge once PRT is complete and I can confirm none of the failures are within the methods you've updated here.

**Comment by mshriver** [comment]
squashing commits since the history isn't very clean and only 2/5 are verified.

**Pull request merged**